src/dcap_provider.cpp

```diff
@@ -1004,3 +1140,386 @@ extern "C" void sgx_free_qe_identity_info(
+                root_ca_crl,
+                p_quote_collateral->root_ca_crl,
+                p_quote_collateral->root_ca_crl_size);
+            if (result == SGX_QL_SUCCESS)
+            {
+                result = fill_qpl_string_buffer(
+                    pck_crl,
+                    p_quote_collateral->pck_crl,
+                    p_quote_collateral->pck_crl_size);
+                if (result == SGX_QL_SUCCESS)
+                {
+                    result = fill_qpl_string_buffer(
+                        tcb_issuer_chain,
+                        p_quote_collateral->tcb_info_issuer_chain,
+                        p_quote_collateral->tcb_info_issuer_chain_size);
+                }
+                if (result == SGX_QL_SUCCESS)
+                {
+                    result = fill_qpl_string_buffer(
+                        tcb_info,
+                        p_quote_collateral->tcb_info,
+                        p_quote_collateral->tcb_info_size);
+                    if (result == SGX_QL_SUCCESS)
+                    {
+                        result = fill_qpl_string_buffer(
+                            qe_identity_issuer_chain,
+                            p_quote_collateral->qe_identity_issuer_chain,
+                            p_quote_collateral->qe_identity_issuer_chain_size);
+                        if (result == SGX_QL_SUCCESS)
+                        {
+                            result = fill_qpl_string_buffer(
+                                qe_identity,
+                                p_quote_collateral->qe_identity,
+                                p_quote_collateral->qe_identity_size);
+                        }
+                    }
+                }
+            }
+        }
+
+        return result;
+    }
+    catch (std::bad_alloc&)
+    {
+        sgx_ql_free_quote_verification_collateral(p_quote_collateral);
+        return SGX_QL_ERROR_OUT_OF_MEMORY;
+    }
+    catch (std::overflow_error& error)
+    {
+        log(SGX_QL_LOG_ERROR, "Overflow error. '%s'", error.what());
+        sgx_ql_free_quote_verification_collateral(p_quote_collateral);
+        return SGX_QL_ERROR_UNEXPECTED;
+    }
+    catch (std::exception& error)
+    {
+        log(SGX_QL_LOG_ERROR,
+            "Unknown exception thrown, error: %s",
+            error.what());
+        return SGX_QL_ERROR_UNEXPECTED;
+    }
+}
+
+extern "C" quote3_error_t sgx_ql_get_qve_identity(
+    char** pp_qve_identity,
+    uint32_t* p_qve_identity_size,
+    char** pp_qve_identity_issuer_chain,
+    uint32_t* p_qve_identity_issuer_chain_size)
+{
+    try
+    {
+        log(SGX_QL_LOG_INFO, "Getting quote verification enclave identity");
+        if (*pp_qve_identity != nullptr)
+        {
+            log(SGX_QL_LOG_ERROR,
+				"Collateral pointer is not null. This memory will be allocated by "
+				"this library");
+            return SGX_QL_ERROR_INVALID_PARAMETER;
+        }
+
+        if (*pp_qve_identity_issuer_chain != nullptr)

```

Comment:
>pp_qve_identity_issuer_chain [](start = 13, length = 28)

Since this is a public API, check 'pp_qve_identity_issuer_chain' first.

Reply:
Thanks! fixed